support py311 in 1.1.3

boldorider4 opened the pull request:
As scikit-learn is a dependency of orange3 whereby the latter has to build for all officially supported py versions, supporting scikit-learn <1.2.0 on py311 becomes necessary, which was otherwise already built for py-38,39,310.

- upstream package changelog has been checked and no breaking change was found for py311
- dependency list has been aligned with that of upstream project to support py311
- a multitude of pytest tests have been found to be failing, mostly due to scikit-learn issues and bugs
- following PRs from upstream project have been taken into consideration: 24141, 24483, 24543 to fix failing tests
- issues with failing tests on osx-64 have been found to be rooted in mkl 2023.1, hence the run_contrained section in the yaml
- no test has been added
- linter check fails have been addressed by adding necessary pinning info, although linter keeps on failing

boldorider4 committed:
update build options and dependencies to support py311

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_6atnrklj7t/clone/recipe/meta.yaml:24: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_6atnrklj7t/clone/recipe/meta.yaml:25: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_6atnrklj7t/clone/recipe/meta.yaml:25: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_6atnrklj7t/clone/recipe/meta.yaml:31: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_6atnrklj7t/clone/recipe/meta.yaml:33: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
WARNING: /tmp/abs_6atnrklj7t/clone/recipe/meta.yaml:84: documentation_overspecified: Using doc_url and doc_source_url is overspecified
WARNING: /tmp/abs_6atnrklj7t/clone/recipe/meta.yaml:84: license_file_overspecified: Using license_file and license_url is overspecified.
Errors were found

</details>

boldorider4 committed:
apply patch from upstream sklearn

The patch seems to solve an issue with tests failing in
model_selection/tests/test_search.

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_83r_t70flx/clone/recipe/meta.yaml:14: patch_unnecessary: patch should not be in build when source/patches is not set
ERROR: /tmp/abs_83r_t70flx/clone/recipe/meta.yaml:26: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_83r_t70flx/clone/recipe/meta.yaml:27: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_83r_t70flx/clone/recipe/meta.yaml:27: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_83r_t70flx/clone/recipe/meta.yaml:33: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_83r_t70flx/clone/recipe/meta.yaml:35: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
WARNING: /tmp/abs_83r_t70flx/clone/recipe/meta.yaml:86: documentation_overspecified: Using doc_url and doc_source_url is overspecified
WARNING: /tmp/abs_83r_t70flx/clone/recipe/meta.yaml:86: license_file_overspecified: Using license_file and license_url is overspecified.
Errors were found

</details>

boldorider4 committed:
move patch dir

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_1alagnrsb2/clone/recipe/meta.yaml:14: patch_unnecessary: patch should not be in build when source/patches is not set
ERROR: /tmp/abs_1alagnrsb2/clone/recipe/meta.yaml:26: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_1alagnrsb2/clone/recipe/meta.yaml:27: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_1alagnrsb2/clone/recipe/meta.yaml:27: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_1alagnrsb2/clone/recipe/meta.yaml:33: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_1alagnrsb2/clone/recipe/meta.yaml:35: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
WARNING: /tmp/abs_1alagnrsb2/clone/recipe/meta.yaml:86: documentation_overspecified: Using doc_url and doc_source_url is overspecified
WARNING: /tmp/abs_1alagnrsb2/clone/recipe/meta.yaml:86: license_file_overspecified: Using license_file and license_url is overspecified.
Errors were found

</details>

boldorider4 committed:
add patches section to recipe

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_85t0_0wy4g/clone/recipe/meta.yaml:28: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_85t0_0wy4g/clone/recipe/meta.yaml:29: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_85t0_0wy4g/clone/recipe/meta.yaml:29: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_85t0_0wy4g/clone/recipe/meta.yaml:35: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_85t0_0wy4g/clone/recipe/meta.yaml:37: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
WARNING: /tmp/abs_85t0_0wy4g/clone/recipe/meta.yaml:88: license_file_overspecified: Using license_file and license_url is overspecified.
WARNING: /tmp/abs_85t0_0wy4g/clone/recipe/meta.yaml:88: documentation_overspecified: Using doc_url and doc_source_url is overspecified
Errors were found

</details>

boldorider4 committed:
apply patch from upstream pr #24483

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_0eydrngg7k/clone/recipe/meta.yaml:29: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_0eydrngg7k/clone/recipe/meta.yaml:30: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_0eydrngg7k/clone/recipe/meta.yaml:30: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_0eydrngg7k/clone/recipe/meta.yaml:36: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_0eydrngg7k/clone/recipe/meta.yaml:38: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
WARNING: /tmp/abs_0eydrngg7k/clone/recipe/meta.yaml:89: license_file_overspecified: Using license_file and license_url is overspecified.
WARNING: /tmp/abs_0eydrngg7k/clone/recipe/meta.yaml:89: documentation_overspecified: Using doc_url and doc_source_url is overspecified
Errors were found

</details>

boldorider4 committed:
restore both patches to git format

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_aah1ca6bvx/clone/recipe/meta.yaml:29: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_aah1ca6bvx/clone/recipe/meta.yaml:30: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_aah1ca6bvx/clone/recipe/meta.yaml:30: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_aah1ca6bvx/clone/recipe/meta.yaml:36: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_aah1ca6bvx/clone/recipe/meta.yaml:38: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
WARNING: /tmp/abs_aah1ca6bvx/clone/recipe/meta.yaml:89: license_file_overspecified: Using license_file and license_url is overspecified.
WARNING: /tmp/abs_aah1ca6bvx/clone/recipe/meta.yaml:89: documentation_overspecified: Using doc_url and doc_source_url is overspecified
Errors were found

</details>

boldorider4 committed:
apply patch from upstream combining pr #24141 and  #24483

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_440meq1lmr/clone/recipe/meta.yaml:28: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_440meq1lmr/clone/recipe/meta.yaml:29: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_440meq1lmr/clone/recipe/meta.yaml:29: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_440meq1lmr/clone/recipe/meta.yaml:35: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_440meq1lmr/clone/recipe/meta.yaml:37: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
WARNING: /tmp/abs_440meq1lmr/clone/recipe/meta.yaml:88: documentation_overspecified: Using doc_url and doc_source_url is overspecified
WARNING: /tmp/abs_440meq1lmr/clone/recipe/meta.yaml:88: license_file_overspecified: Using license_file and license_url is overspecified.
Errors were found

</details>

boldorider4 committed:
remove unused patches

boldorider4 committed:
ignore linking of std c++lib on osx-arm64

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_e0xi2lump3/clone/recipe/meta.yaml:30: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_e0xi2lump3/clone/recipe/meta.yaml:31: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_e0xi2lump3/clone/recipe/meta.yaml:31: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_e0xi2lump3/clone/recipe/meta.yaml:37: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_e0xi2lump3/clone/recipe/meta.yaml:39: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
WARNING: /tmp/abs_e0xi2lump3/clone/recipe/meta.yaml:90: license_file_overspecified: Using license_file and license_url is overspecified.
WARNING: /tmp/abs_e0xi2lump3/clone/recipe/meta.yaml:90: documentation_overspecified: Using doc_url and doc_source_url is overspecified
Errors were found

</details>

boldorider4 committed:
ignore linking of std c++lib on osx-arm64

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_79kpsg5trl/clone/recipe/meta.yaml:31: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_79kpsg5trl/clone/recipe/meta.yaml:32: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_79kpsg5trl/clone/recipe/meta.yaml:32: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_79kpsg5trl/clone/recipe/meta.yaml:38: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_79kpsg5trl/clone/recipe/meta.yaml:40: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
WARNING: /tmp/abs_79kpsg5trl/clone/recipe/meta.yaml:91: license_file_overspecified: Using license_file and license_url is overspecified.
WARNING: /tmp/abs_79kpsg5trl/clone/recipe/meta.yaml:91: documentation_overspecified: Using doc_url and doc_source_url is overspecified
Errors were found

</details>

boldorider4 committed:
fix patch header

boldorider4 committed:
apply patch from pr 24543 to fix last test failures

boldorider4 committed:
update patch section in recipe

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_69wlts6e9l/clone/recipe/meta.yaml:33: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_69wlts6e9l/clone/recipe/meta.yaml:34: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_69wlts6e9l/clone/recipe/meta.yaml:34: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_69wlts6e9l/clone/recipe/meta.yaml:40: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_69wlts6e9l/clone/recipe/meta.yaml:42: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
WARNING: /tmp/abs_69wlts6e9l/clone/recipe/meta.yaml:93: license_file_overspecified: Using license_file and license_url is overspecified.
WARNING: /tmp/abs_69wlts6e9l/clone/recipe/meta.yaml:93: documentation_overspecified: Using doc_url and doc_source_url is overspecified
Errors were found

</details>

boldorider4 committed:
rearranging patches to retain gh pr history

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_22fvb1bp7v/clone/recipe/meta.yaml:33: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_22fvb1bp7v/clone/recipe/meta.yaml:34: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_22fvb1bp7v/clone/recipe/meta.yaml:34: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_22fvb1bp7v/clone/recipe/meta.yaml:40: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_22fvb1bp7v/clone/recipe/meta.yaml:42: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
WARNING: /tmp/abs_22fvb1bp7v/clone/recipe/meta.yaml:93: license_file_overspecified: Using license_file and license_url is overspecified.
WARNING: /tmp/abs_22fvb1bp7v/clone/recipe/meta.yaml:93: documentation_overspecified: Using doc_url and doc_source_url is overspecified
Errors were found

</details>

boldorider4 committed:
fix patch indeces

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_50pm8p1jp1/clone/recipe/meta.yaml:33: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_50pm8p1jp1/clone/recipe/meta.yaml:34: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_50pm8p1jp1/clone/recipe/meta.yaml:34: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_50pm8p1jp1/clone/recipe/meta.yaml:40: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_50pm8p1jp1/clone/recipe/meta.yaml:42: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
WARNING: /tmp/abs_50pm8p1jp1/clone/recipe/meta.yaml:93: license_file_overspecified: Using license_file and license_url is overspecified.
WARNING: /tmp/abs_50pm8p1jp1/clone/recipe/meta.yaml:93: documentation_overspecified: Using doc_url and doc_source_url is overspecified
Errors were found

</details>

boldorider4 committed:
fix problem with multiple commits patches

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_fbp4qxbw7h/clone/recipe/meta.yaml:41: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_fbp4qxbw7h/clone/recipe/meta.yaml:42: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_fbp4qxbw7h/clone/recipe/meta.yaml:42: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_fbp4qxbw7h/clone/recipe/meta.yaml:48: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_fbp4qxbw7h/clone/recipe/meta.yaml:50: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
WARNING: /tmp/abs_fbp4qxbw7h/clone/recipe/meta.yaml:101: documentation_overspecified: Using doc_url and doc_source_url is overspecified
WARNING: /tmp/abs_fbp4qxbw7h/clone/recipe/meta.yaml:101: license_file_overspecified: Using license_file and license_url is overspecified.
Errors were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_db2r7r0umx/clone/recipe/meta.yaml:41: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_db2r7r0umx/clone/recipe/meta.yaml:42: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_db2r7r0umx/clone/recipe/meta.yaml:42: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_db2r7r0umx/clone/recipe/meta.yaml:48: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_db2r7r0umx/clone/recipe/meta.yaml:50: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
WARNING: /tmp/abs_db2r7r0umx/clone/recipe/meta.yaml:101: license_file_overspecified: Using license_file and license_url is overspecified.
WARNING: /tmp/abs_db2r7r0umx/clone/recipe/meta.yaml:101: documentation_overspecified: Using doc_url and doc_source_url is overspecified
Errors were found

</details>

boldorider4 committed:
attempt to get pytest-xdist to succeed

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_b0aw4bqtqq/clone/recipe/meta.yaml:41: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_b0aw4bqtqq/clone/recipe/meta.yaml:42: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_b0aw4bqtqq/clone/recipe/meta.yaml:42: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_b0aw4bqtqq/clone/recipe/meta.yaml:48: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_b0aw4bqtqq/clone/recipe/meta.yaml:50: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
WARNING: /tmp/abs_b0aw4bqtqq/clone/recipe/meta.yaml:101: documentation_overspecified: Using doc_url and doc_source_url is overspecified
WARNING: /tmp/abs_b0aw4bqtqq/clone/recipe/meta.yaml:101: license_file_overspecified: Using license_file and license_url is overspecified.
Errors were found

</details>

boldorider4 committed:
attempt to solve mkl problem with numpy

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_ddnvsnjpvr/clone/recipe/meta.yaml:41: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_ddnvsnjpvr/clone/recipe/meta.yaml:42: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_ddnvsnjpvr/clone/recipe/meta.yaml:42: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_ddnvsnjpvr/clone/recipe/meta.yaml:48: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_ddnvsnjpvr/clone/recipe/meta.yaml:50: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
WARNING: /tmp/abs_ddnvsnjpvr/clone/recipe/meta.yaml:103: documentation_overspecified: Using doc_url and doc_source_url is overspecified
WARNING: /tmp/abs_ddnvsnjpvr/clone/recipe/meta.yaml:103: license_file_overspecified: Using license_file and license_url is overspecified.
Errors were found

</details>

boldorider4 committed:
make specific pinning for non-python dep

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_29629_z_g8/clone/recipe/meta.yaml:41: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_29629_z_g8/clone/recipe/meta.yaml:42: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_29629_z_g8/clone/recipe/meta.yaml:48: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_29629_z_g8/clone/recipe/meta.yaml:50: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
WARNING: /tmp/abs_29629_z_g8/clone/recipe/meta.yaml:103: documentation_overspecified: Using doc_url and doc_source_url is overspecified
WARNING: /tmp/abs_29629_z_g8/clone/recipe/meta.yaml:103: license_file_overspecified: Using license_file and license_url is overspecified.
Errors were found

</details>

boldorider4 committed:
do not pin specific versions if not necessary for python packages

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_eftyliwep1/clone/recipe/meta.yaml:41: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_eftyliwep1/clone/recipe/meta.yaml:42: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_eftyliwep1/clone/recipe/meta.yaml:48: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_eftyliwep1/clone/recipe/meta.yaml:50: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
WARNING: /tmp/abs_eftyliwep1/clone/recipe/meta.yaml:103: documentation_overspecified: Using doc_url and doc_source_url is overspecified
WARNING: /tmp/abs_eftyliwep1/clone/recipe/meta.yaml:103: license_file_overspecified: Using license_file and license_url is overspecified.
Errors were found

</details>

M-Waszkiewicz-Anaconda commented on recipe/meta.yaml:
```diff
@@ -7,37 +7,59 @@ package:
 source:
   url: https://github.com/scikit-learn/scikit-learn/archive/{{ version }}.tar.gz
   sha256: b7c0a9fb8dfcefcfc7ef8fe02a064d194980251d57efaea972cb76005afb3c63
+  patches:
```
what is the scope of this ticket ? just add py 3.11 support for existing 1.1.3 version, if so does these patches are needed for older python versions  ?,  it has built previously, is this still can be build for py < 3.11 ? If so maybe those patches should be applied to py 3.11 only ?

boldorider4 replied:
The package builds for all python versions. Those patches are required for a variety of tests to run succefully. References to PRs that fix those tests have been included in this PR description.
I did not check whether those tests also fail on py != 311, but since those issues that the referenced PRs fix were acknowledged as issues that must be fixed, I think applying these patches to all py version does no harm.

M-Waszkiewicz-Anaconda commented on recipe/meta.yaml:
```diff
@@ -7,37 +7,59 @@ package:
+   - patches/sklearn-pr24543-17035b.patch
 
 build:
   number: 0
```
bump up

M-Waszkiewicz-Anaconda commented on recipe/meta.yaml:
```diff
@@ -7,37 +7,59 @@ package:
-  script: python -m pip install --no-deps --ignore-installed .  --verbose
+  script: python -m pip install . --no-deps --no-build-isolation --ignore-installed -vv
   skip: true  # [py<38]
+  missing_dso_whitelist:
```
is this needed for py 3.11 only ?

M-Waszkiewicz-Anaconda commented on recipe/meta.yaml:
```diff
@@ -7,37 +7,59 @@ package:
+    - llvm-openmp 14.0.6  # [osx]
     - numpy   1.19  # [py==38 or py==39]
     - numpy   1.21  # [py==310]
+    - numpy   1.23  # [py>=311]
```
please compare with conda_build_config.yaml it is almost the same, can it be reused ?

M-Waszkiewicz-Anaconda commented on recipe/meta.yaml:
```diff
@@ -63,7 +85,7 @@ test:
     - export OPENBLAS_NUM_THREADS=1       # [not win]
     - export OMP_NUM_THREADS=1            # [not win]
-    - pytest --timeout 300 -n {{ test_cpus }} --verbose --pyargs sklearn -k "not ({{ tests_to_skip }})"
+    - pytest --max-worker-restart 10 --timeout 300 -n {{ test_cpus }} --verbose --pyargs sklearn -k "not ({{ tests_to_skip }})"
```
what is this new param for ?

boldorider4 replied:
thanks for catching this! it's a leftover of some experiments. Will remove.

akabanovs commented on recipe/meta.yaml:
```diff
@@ -7,37 +7,59 @@ package:
+    - m2-patch # [win]
   host:
     - python
     - cython >=0.29.24
```
Can use 0.29.32 which is available for py >=3.8, <=3.11 on all archs.

akabanovs commented on recipe/meta.yaml:
```diff
@@ -7,37 +7,59 @@ package:
+    - joblib
+    - llvm-openmp 14.0.6  # [osx]
     - numpy   1.19  # [py==38 or py==39]
     - numpy   1.21  # [py==310]
```
I think it asks for 1.22.3 for py 3.10 on windows?
https://github.com/scikit-learn/scikit-learn/blob/1.1.3/pyproject.toml#L17

cbouss replied:
This looks fine to me. That would be taken care of by the scipy package itself.
When using oldest-supported-numpy it refers to what is available on pypi, which may be different that what we have on main.

akabanovs reviewed (commented):
doc_source_url and license_url is over-specification and not strictly required.

akabanovs commented on recipe/meta.yaml:
```diff
@@ -7,37 +7,59 @@ package:
-    - scipy >=1.3.2
-    - threadpoolctl >=2.0.0
+    - scipy >=1.3.2,<=1.9.3 # [py<311]
+    - scipy 1.10.0          # [py>=311]
```
Why are there different pinnings here? I mean I can't see it in the upstream, but py 3.11 is also not a target version in the upstream.

cbouss replied:
same comment, why not use scipy 1.9.3 on 3.11? It's also available. Or does it conflict with another package?

boldorider4 committed:
restrict patches to py311

boldorider4 committed:
bump build version

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_61b037psb2/clone/recipe/meta.yaml:19: patch_unnecessary: patch should not be in build when source/patches is not set
ERROR: /tmp/abs_61b037psb2/clone/recipe/meta.yaml:31: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_61b037psb2/clone/recipe/meta.yaml:32: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_61b037psb2/clone/recipe/meta.yaml:38: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_61b037psb2/clone/recipe/meta.yaml:40: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
WARNING: /tmp/abs_61b037psb2/clone/recipe/meta.yaml:93: license_file_overspecified: Using license_file and license_url is overspecified.
WARNING: /tmp/abs_61b037psb2/clone/recipe/meta.yaml:93: documentation_overspecified: Using doc_url and doc_source_url is overspecified
Errors were found

</details>

boldorider4 committed:
fix selector

boldorider4 committed:
remove useless switch

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_36a2w4j7bh/clone/recipe/meta.yaml:30: patch_unnecessary: patch should not be in build when source/patches is not set
ERROR: /tmp/abs_36a2w4j7bh/clone/recipe/meta.yaml:42: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_36a2w4j7bh/clone/recipe/meta.yaml:43: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_36a2w4j7bh/clone/recipe/meta.yaml:49: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_36a2w4j7bh/clone/recipe/meta.yaml:51: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
WARNING: /tmp/abs_36a2w4j7bh/clone/recipe/meta.yaml:104: documentation_overspecified: Using doc_url and doc_source_url is overspecified
WARNING: /tmp/abs_36a2w4j7bh/clone/recipe/meta.yaml:104: license_file_overspecified: Using license_file and license_url is overspecified.
Errors were found

</details>

boldorider4 committed:
upgrade cython pinning

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_e2n50jg_ey/clone/recipe/meta.yaml:30: patch_unnecessary: patch should not be in build when source/patches is not set
ERROR: /tmp/abs_e2n50jg_ey/clone/recipe/meta.yaml:42: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_e2n50jg_ey/clone/recipe/meta.yaml:43: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_e2n50jg_ey/clone/recipe/meta.yaml:49: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_e2n50jg_ey/clone/recipe/meta.yaml:51: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
WARNING: /tmp/abs_e2n50jg_ey/clone/recipe/meta.yaml:104: license_file_overspecified: Using license_file and license_url is overspecified.
WARNING: /tmp/abs_e2n50jg_ey/clone/recipe/meta.yaml:104: documentation_overspecified: Using doc_url and doc_source_url is overspecified
Errors were found

</details>

boldorider4 committed:
fix linter warnings

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_84go_27pu1/clone/recipe/meta.yaml:30: patch_unnecessary: patch should not be in build when source/patches is not set
ERROR: /tmp/abs_84go_27pu1/clone/recipe/meta.yaml:42: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_84go_27pu1/clone/recipe/meta.yaml:43: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_84go_27pu1/clone/recipe/meta.yaml:49: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_84go_27pu1/clone/recipe/meta.yaml:51: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>

boldorider4 committed:
fix pinnings in the host section

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_50f8r7gccy/clone/recipe/meta.yaml:30: patch_unnecessary: patch should not be in build when source/patches is not set
ERROR: /tmp/abs_50f8r7gccy/clone/recipe/meta.yaml:42: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_50f8r7gccy/clone/recipe/meta.yaml:43: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_50f8r7gccy/clone/recipe/meta.yaml:49: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_50f8r7gccy/clone/recipe/meta.yaml:51: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>

boldorider4 committed:
add pinning comments

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_b6e_7xr9ps/clone/recipe/meta.yaml:30: patch_unnecessary: patch should not be in build when source/patches is not set
ERROR: /tmp/abs_b6e_7xr9ps/clone/recipe/meta.yaml:42: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_b6e_7xr9ps/clone/recipe/meta.yaml:43: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_b6e_7xr9ps/clone/recipe/meta.yaml:49: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_b6e_7xr9ps/clone/recipe/meta.yaml:51: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>

M-Waszkiewicz-Anaconda commented on recipe/meta.yaml:
```diff
@@ -7,37 +7,62 @@ package:
+    - numpy   1.23  # [py>=311]
     - pip
-    - scipy >=1.3.2
+    - scipy >=1.3.2 # [py<311]
```
strict pinning ?

boldorider4 committed:
fix range pinning in host section

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_43qzb8wnkh/clone/recipe/meta.yaml:30: patch_unnecessary: patch should not be in build when source/patches is not set
ERROR: /tmp/abs_43qzb8wnkh/clone/recipe/meta.yaml:42: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_43qzb8wnkh/clone/recipe/meta.yaml:43: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_43qzb8wnkh/clone/recipe/meta.yaml:51: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>

M-Waszkiewicz-Anaconda reviewed (approved):
LGTM

akabanovs reviewed (approved):
LGTM

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_f961iv4acq/clone/recipe/meta.yaml:30: patch_unnecessary: patch should not be in build when source/patches is not set
ERROR: /tmp/abs_f961iv4acq/clone/recipe/meta.yaml:42: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_f961iv4acq/clone/recipe/meta.yaml:43: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_f961iv4acq/clone/recipe/meta.yaml:51: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>

varlackc reviewed (approved):
LGTM

boldorider4 committed:
apply patch section only to >=py311

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_a3c3imcvr7/clone/recipe/meta.yaml:30: patch_unnecessary: patch should not be in build when source/patches is not set
ERROR: /tmp/abs_a3c3imcvr7/clone/recipe/meta.yaml:42: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_a3c3imcvr7/clone/recipe/meta.yaml:43: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_a3c3imcvr7/clone/recipe/meta.yaml:51: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>

boldorider4 committed:
depend on patch only for >=py311

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_0880h0wb0n/clone/recipe/meta.yaml:42: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_0880h0wb0n/clone/recipe/meta.yaml:43: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_0880h0wb0n/clone/recipe/meta.yaml:51: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_78hvl0a6_q/clone/recipe/meta.yaml:42: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_78hvl0a6_q/clone/recipe/meta.yaml:43: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_78hvl0a6_q/clone/recipe/meta.yaml:51: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>
